SQRLDotNetClientUI/Platform/OSX/SystemEventNotifier.cs

```diff
@@ -0,0 +1,139 @@
+﻿using Avalonia;
+using MonoMac.AppKit;
+using MonoMac.Foundation;
+using Serilog;
+using SQRLDotNetClientUI.Models;
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SQRLDotNetClientUI.Platform.OSX
+{
+    /// <summary>
+    /// Provides access to Windows system events that are relevant to the 
+    /// clearing of QuickPass-related data from RAM, like entering
+    /// an idle state, logging off or entering a sleep or hilbernation
+    /// state etc.
+    /// 
+    /// Most of the events are gathered by creating a native Apple App Delegate (at App Initialize) and 
+    /// registering and listening to event related window messages. Some
+    /// events require polling, so a low-impact polling thread is implemented
+    /// to peridodically check for those events.
+    /// </summary>
+    public class SystemEventNotifier : ISystemEventNotifier
+    {
+
+        /// <summary>
+        /// Specifies the checking interval for the polling thread.
+        /// </summary>
+        private readonly int POLL_INTERVAL = 5000;
+
+
+        private CancellationTokenSource _cts = null;
```

Comment:
You're not instantiating the `CancellationTokenSource` and you are not assigning the `CancellationToken`.
This should be part of your constructor:

```C#
_cts = new CancellationTokenSource();
_ct = _cts.Token;
```

Additionally, I would also add the destructor that cancels the polling task before garbage-collecting the object:

```C#
~SystemEventNotifier()
{
    // Cancel the polling task
    _cts.Cancel();
}
```